File: Maps_Project.Web/Modules/Membership/Account/SignUp/SignUpForm.cs
namespace Maps_Project.Membership;

[FormScript("Membership.SignUp")]
public class SignUpForm
{
    [Required(true), Placeholder("full name")]
    public string DisplayName { get; set; }
    [EmailAddressEditor, Required(true), Placeholder("e-mail")]
    public string Email { get; set; }
    [EmailAddressEditor, Required(true), Placeholder("confirm email")]
    public string ConfirmEmail { get; set; }
    [PasswordEditor, Required(true), Placeholder("password")]
    public string Password { get; set; }
    [PasswordEditor, Required(true), Placeholder("confirm password")]
    public string ConfirmPassword { get; set; }
    // make sure you have configured Recaptcha:SiteKey and Recaptcha:SecretKey in appsettings.json
    [DisplayName(""), Recaptcha]
    public string Recaptcha { get; set; }
}

File: Maps_Project.Web/Modules/ServerTypes/Membership/SignUpForm.ts
import { StringEditor, EmailAddressEditor, PasswordEditor, Widget, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SignUpForm {
    DisplayName: StringEditor;
    Email: EmailAddressEditor;
    ConfirmEmail: EmailAddressEditor;
    Password: PasswordEditor;
    ConfirmPassword: PasswordEditor;
    Recaptcha: Widget;
}

export class SignUpForm extends PrefixedContext {
    static readonly formKey = 'Membership.SignUp';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SignUpForm.init)  {
            SignUpForm.init = true;

            var w0 = StringEditor;
            var w1 = EmailAddressEditor;
            var w2 = PasswordEditor;
            var w3 = Widget;

            initFormType(SignUpForm, [
                'DisplayName', w0,
                'Email', w1,
                'ConfirmEmail', w1,
                'Password', w2,
                'ConfirmPassword', w2,
                'Recaptcha', w3
            ]);
        }
    }
}